Server/mods/deathmatch/logic/CPedSync.cpp

```diff
@@ -191,73 +187,148 @@ void CPedSync::Packet_PedSync(CPedSyncPacket& Packet)
 {
     // Grab the player
     CPlayer* pPlayer = Packet.GetSourcePlayer();
-    if (pPlayer && pPlayer->IsJoined())
+    if (!pPlayer || !pPlayer->IsJoined())
+        return;
+
+    // Grab the tick count
+    long long llTickCountNow = GetModuleTickCount64();
+
+    // Apply the data for each ped in the packet
+    for (auto iter = Packet.IterBegin(); iter != Packet.IterEnd(); iter++)
     {
-        // Apply the data for each ped in the packet
-        vector<CPedSyncPacket::SyncData*>::const_iterator iter = Packet.IterBegin();
-        for (; iter != Packet.IterEnd(); iter++)
+        CPedSyncPacket::SyncData* pData = *iter;
+
+        // Grab the ped this packet is for
+        CElement* pPedElement = CElementIDs::GetElement(pData->Model);
+        if (!pPedElement || !IS_PED(pPedElement))
```

Comment:
Perhaps add a dassert as well? Might be helpful.

Reply:
Could be
But it's not really relevant IMO, it would happen expectedly when a ped is destroyed but some sync comes in because the client doesnt know yet. (must happen often with high latency)